packages/did-resolver/src/memory-cache.ts

```diff
@@ -0,0 +1,48 @@
+import { DAY } from '@atproto/common-web'
+import { DidCache } from './did-cache'
+import { CacheResult, DidDocument } from './types'
+
+type CacheVal = {
+  doc: DidDocument
+  updatedAt: number
+}
+
+export class MemoryCache extends DidCache {
+  public ttl: number
+  constructor(ttl?: number) {
+    super()
+    this.ttl = ttl ?? DAY
+  }
+
+  public cache: Record<string, CacheVal> = {}
```

Comment:
A `Map` may be better suited to backing this cache (even though the keys are strings anyway), particularly if we start removing items from the cache.

Reply:
Also slightly nicer from a typing perspective I think, since it knows `get()` may come back `undefined`, but `cache[did]` is always considered to be present.

Reply:
yup yup good call switched it over 👌